packages/astro/src/core/add/index.ts

```diff
@@ -0,0 +1,456 @@
+
+	ensureImport(ast, t.importDeclaration([t.importDefaultSpecifier(integrationId)], t.stringLiteral(integration.packageName)));
+
+	visit(ast, {
+		// eslint-disable-next-line @typescript-eslint/no-shadow
+		ExportDefaultDeclaration(path) {
+			if (!t.isCallExpression(path.node.declaration)) return;
+
+			const configObject = path.node.declaration.arguments[0];
+			if (!t.isObjectExpression(configObject)) return;
+
+			let integrationsProp = configObject.properties.find((prop) => {
+				if (prop.type !== 'ObjectProperty') return false;
+				if (prop.key.type === 'Identifier') {
+					if (prop.key.name === 'integrations') return true;
+				}
+				if (prop.key.type === 'StringLiteral') {
+					if (prop.key.value === 'integrations') return true;
+				}
+				return false;
+			}) as t.ObjectProperty | undefined;
+
+			const integrationCall = t.callExpression(integrationId, []);
+
+			if (!integrationsProp) {
+				configObject.properties.push(t.objectProperty(t.identifier('integrations'), t.arrayExpression([integrationCall])));
+				return;
+			}
+
+			if (integrationsProp.value.type !== 'ArrayExpression') throw new Error('Unable to parse integrations');
+
+			const existingIntegrationCall = integrationsProp.value.elements.find(
+				(expr) => t.isCallExpression(expr) && t.isIdentifier(expr.callee) && expr.callee.name === integrationId.name
+			);
+
+			if (existingIntegrationCall) return;
+
+			integrationsProp.value.elements.push(integrationCall);
+		},
+	});
+}
+
+const enum UpdateResult {
+	none,
+	updated,
+	cancelled,
+	failure,
+}
+
+async function updateAstroConfig({ configURL, ast, flags, logging }: { configURL: URL; ast: t.File; flags: yargs.Arguments; logging: LogOptions }): Promise<UpdateResult> {
+	const input = await fs.readFile(fileURLToPath(configURL), { encoding: 'utf-8' });
+	let output = await generate(ast);
+	const comment = '// https://astro.build/config';
+	const defaultExport = 'export default defineConfig';
+	output = output.replace(` ${comment}`, '');
+	output = output.replace(`${defaultExport}`, `\n${comment}\n${defaultExport}`);
+
+	if (input === output) {
+		return UpdateResult.none;
+	}
+
+	let changes = [];
+	for (const change of diffWords(input, output)) {
+		let lines = change.value
+			.trim()
+			.split('\n')
+			.slice(0, change.count)
+		if (lines.length === 0) continue;
+		if (change.added) {
+			if (!change.value.trim()) continue;
+			changes.push(change.value);
+		}
+	}
+	if (changes.length === 0) {
+		return UpdateResult.none;
+	}
+
+	let diffed = output;
+	for (let newContent of changes) {
+		const coloredOutput = newContent
+			.split('\n')
+			.map((ln) => (ln ? green(ln) : ''))
+			.join('\n');
+		diffed = diffed.replace(newContent, coloredOutput);
+	}
+
+	const message = `\n${boxen(diffed, { margin: 0.5, padding: 0.5, borderStyle: 'round', title: configURL.pathname.split('/').pop() })}\n`;
+
+	info(logging, null, `\n  ${magenta('Astro will update your configuration with these changes...')}\n${message}`);
```

Comment:
nit: `...` vs. `:`? also `these changes` vs. `the following changes`?

Slight preference for `Astro will make the following changes to your config file:`

Reply:
This matches the `Astro will run the following command to install...` message below in format as well, which is a happy coincidence